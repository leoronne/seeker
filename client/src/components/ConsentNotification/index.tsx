import React from 'react';
import CookieConsent, { Cookies } from 'react-cookie-consent';
import { useTranslation } from 'react-i18next';

import { Container } from './styles';

const ConsentNotification: React.FC = () => {
  const { t } = useTranslation();

  if (Cookies.get('@square-landing-page:cookies') === 'true') return null;

  return (
    <Container>
      <CookieConsent
        location="bottom"
        buttonText={t('accept')}
        cookieName="enableCookies"
        expires={150}
        onAccept={() => {
          Cookies.set('@square-landing-page:cookies', true);
        }}
        debug
      >
        <h2>{t('cookie1')}</h2>
        <span className="cookie-subtitle">{t('cookie2')}</span>
      </CookieConsent>
    </Container>
  );
};

export default ConsentNotification;
