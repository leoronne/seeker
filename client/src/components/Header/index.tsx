import React from 'react';
import { Link } from 'react-router-dom';
import { MenuItem } from '@material-ui/core';

import { useLanguage } from '../../hooks/useLanguage';

import { useStyles, Select } from '../../styles/MaterialUI';

import { ReactComponent as BrazilIcon } from '../../assets/svg/brazil.svg';
import { ReactComponent as USAIcon } from '../../assets/svg/usa.svg';

import logo from '../../assets/svg/logo-horizontal.svg';

import { HeaderWrapper, HeaderLeft, HeaderRight } from './styles';

const Header: React.FC = () => {
  const { language, changeLanguage } = useLanguage();
  const classes = useStyles();

  return (
    <HeaderWrapper>
      <HeaderLeft>
        <h1>
          <Link to="/">
            <img src={logo} alt="Seeker" />
          </Link>
        </h1>
      </HeaderLeft>

      <HeaderRight>
        <Select
          value={language}
          onChange={e => changeLanguage(String(e.target.value))}
          MenuProps={{ classes: { paper: classes.selectOptions } }}
          inputProps={{ 'data-testid': 'language-select' }}
        >
          <MenuItem value="en">
            <USAIcon />
            <span>English (United States)</span>
          </MenuItem>
          <MenuItem value="pt">
            <BrazilIcon />
            <span>PortuguÃªs (Brasil)</span>
          </MenuItem>
        </Select>
      </HeaderRight>
    </HeaderWrapper>
  );
};

export default Header;
