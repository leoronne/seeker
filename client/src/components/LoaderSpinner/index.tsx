import React from 'react';
import { CircularProgress } from '@material-ui/core';

import { Container } from './styles';

const Loader: React.FC<{ color: string }> = ({ color }) => {
  return (
    <Container className="loader-container">
      <CircularProgress size={15} style={{ color }} />
    </Container>
  );
};
export default Loader;
