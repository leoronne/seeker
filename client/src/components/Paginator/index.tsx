/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable jsx-a11y/no-static-element-interactions */
import React from 'react';
import { useMediaQuery } from '@material-ui/core';

import { Container } from './styles';

function paginate(totalItems: number, currentPage = 1, pageSize = 15, maxPages = 15) {
  // calculate total pages
  const totalPages = Math.ceil(totalItems / pageSize);

  let actualPage = currentPage;

  // ensure current page isn't out of range
  if (actualPage < 1) {
    actualPage = 1;
  } else if (actualPage > totalPages) {
    actualPage = totalPages;
  }

  let startPage: number;
  let endPage: number;
  if (totalPages <= maxPages) {
    // total pages less than max so show all pages
    startPage = 1;
    endPage = totalPages;
  } else {
    // total pages more than max so calculate start and end pages
    const maxPagesBeforeCurrentPage = Math.floor(maxPages / 2);
    const maxPagesAfterCurrentPage = Math.ceil(maxPages / 2) - 1;
    if (actualPage <= maxPagesBeforeCurrentPage) {
      // current page near the start
      startPage = 1;
      endPage = maxPages;
    } else if (actualPage + maxPagesAfterCurrentPage >= totalPages) {
      // current page near the end
      startPage = totalPages - maxPages + 1;
      endPage = totalPages;
    } else {
      // current page somewhere in the middle
      startPage = actualPage - maxPagesBeforeCurrentPage;
      endPage = actualPage + maxPagesAfterCurrentPage;
    }
  }

  // calculate start and end item indexes
  const startIndex = (actualPage - 1) * pageSize;
  const endIndex = Math.min(startIndex + pageSize - 1, totalItems - 1);

  // create an array of pages to ng-repeat in the pager control
  const pages = Array.from(Array(endPage + 1 - startPage).keys()).map(i => startPage + i);

  // return object with all pager properties required by the view
  return {
    totalItems,
    currentPage,
    pageSize,
    totalPages,
    startPage,
    endPage,
    startIndex,
    endIndex,
    pages,
  };
}

interface Props {
  totalRecords: number;
  recordsPerPage: number;
  setOffset: React.Dispatch<React.SetStateAction<number>>;
  currentPage: number;
  setCurrentPage: (page: number) => Promise<void>;
}

const Paginator: React.FC<Props> = ({ totalRecords = 1, recordsPerPage = 1, currentPage = 1, setCurrentPage }) => {
  const width = useMediaQuery('(max-width: 568px)');

  const maxPages = width ? 2 : 5;

  const pages = paginate(totalRecords, currentPage, recordsPerPage, maxPages);

  let lines = [];

  if (pages.totalPages > 1) {
    lines = pages.pages.map(page => {
      return (
        <div
          key={page}
          className={page === currentPage ? 'current' : 'page'}
          onClick={() => setCurrentPage(page)}
          data-testid={`${page === currentPage ? 'page-active' : `page-${page}`}`}
        >
          {page}
        </div>
      );
    });
  } else {
    lines.push(
      <div className="current" onClick={() => setCurrentPage(1)} key={1}>
        1
      </div>
    );
  }

  return (
    <Container>
      <div
        className={`page${currentPage === 1 ? ' page-disabled' : ''}`}
        onClick={() => {
          if (currentPage > 1 && pages.totalPages > 0) setCurrentPage(currentPage - 1);
        }}
        data-testid="previous-page"
      >
        {'<'}
      </div>
      {lines}
      <div
        className={`page${pages.totalPages === currentPage || pages.totalPages === 0 ? ' page-disabled' : ''}`}
        onClick={() => {
          if (pages.totalPages !== currentPage && pages.totalPages > 0) setCurrentPage(currentPage + 1);
        }}
        data-testid="next-page"
      >
        {'>'}
      </div>
    </Container>
  );
};

export default Paginator;
