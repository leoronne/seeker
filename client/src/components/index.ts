export { default as Header } from './Header';
export { default as LoaderSpinner } from './LoaderSpinner';
export { default as ConsentNotification } from './ConsentNotification';
export { default as Paginator } from './Paginator';
