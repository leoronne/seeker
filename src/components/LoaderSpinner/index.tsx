import React from 'react';
import { CircularProgress } from '@material-ui/core';

import { Container } from './styles';

const Loader: React.FC = () => {
  return (
    <Container className="loader-container">
      <CircularProgress size={15} style={{ color: '#ea3d3d' }} />
    </Container>
  );
};
export default Loader;
