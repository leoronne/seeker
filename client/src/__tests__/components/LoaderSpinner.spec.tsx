import React from 'react';
import { render } from '@testing-library/react';

import { LoaderSpinner } from '../../components';

describe('LoaderSpinner', () => {
  it('should render', async () => {
    const { asFragment } = render(<LoaderSpinner color="#fff" />);

    const html = asFragment();

    expect(html).toMatchSnapshot();
  });
});
