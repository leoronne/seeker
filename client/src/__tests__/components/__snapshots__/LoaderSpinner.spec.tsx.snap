// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoaderSpinner should render 1`] = `
<DocumentFragment>
  <div
    class="sc-hKFyIo deuFTM loader-container"
  >
    <div
      class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
      role="progressbar"
      style="width: 15px; height: 15px; color: rgb(255, 255, 255);"
    >
      <svg
        class="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;
